apps/tangle-dapp/data/NominationsPayouts/usePayouts.ts

```diff
@@ -31,78 +31,78 @@ type PayoutData = {
 
 export default function usePayouts(): PayoutData {
   const { isLoading, data: payouts } = usePayoutsStore();
-
   const { setWithPreviousValue: setCachedPayouts } = useLocalStorage(
     LocalStorageKey.PAYOUTS,
     true,
   );
-
   const { rpcEndpoint, network } = useNetworkStore();
-
   const activeSubstrateAddress = useSubstrateAddress();
 
   const { data: eraTotalRewards } = useEraTotalRewards();
-
   const { result: validators } = useApiRx(
     useCallback((api) => api.query.staking.validators.entries(), []),
   );
-
   const mappedValidatorInfo = useMemo(() => {
     const map = new Map<string, PalletStakingValidatorPrefs>();
-
     validators?.forEach(([storageKey, validatorInfo]) => {
       map.set(storageKey.args[0].toString(), validatorInfo);
     });
-
     return map;
   }, [validators]);
-
   const unclaimedRewards = useNominationsUnclaimedRewards();
+  const hasComputedPayoutsRef = useRef(false);
+
+  useEffect(() => {
+    // Make sure all data is available before computing payouts
+    if (
+      activeSubstrateAddress === null ||
+      unclaimedRewards.length === 0 ||
+      eraTotalRewards === null ||
+      eraTotalRewards.size === 0 ||
+      mappedValidatorInfo.size === 0 ||
+      hasComputedPayoutsRef.current ||
```

Comment:
Why do you think we need a `hasComputedPayoutsRef` here instead of checking if `payouts.length > 0`?